misc/emacs/ql-mode-base.el

```diff
@@ -0,0 +1,74 @@
+;;; ql-mode-base.el --- A major mode for editing QL files
+
+;;; Commentary:
+;;
+;; A basic major mode for editing QL files, a more advanced major
+;; mode is available for internal use at Semmle.
+;;
+;; Provides syntax highlightning, comment support, and a mode-specific
+;; keymap.
+
+;;; Code:
+
+(defconst ql--at-type-regex "\\_<@\\w+\\>")
+(defconst ql--predicate-regex "\\(\\_<\\w+\\(\\+\\|\\*\\)?\\_>\\)\\s-*(")
+(defconst ql--primitive-type-regex (regexp-opt '("int" "string" "float" "boolean" "date") 'symbols))
+(defconst ql--annotation-regex (regexp-opt '("abstract" "cached" "external" "final" "transient" "library" "private" "deprecated" "override" "query" "language" "bindingset") 'words))
```

Comment:
It probably makes sense to keep this consistent with the [vim list](https://github.com/Semmle/ql/pull/871/files#diff-feccfb634405a05309579436eefc55e0R2). This list additionally has `transient`, while vim has:

```
monotonicAggregates
noinline
nomagic
noopt
pragma
```

Reply:
Fair, I have changed it such that we now highlight all arguments to the parameterized annotations `pragma`, `language` and `bindingset`. In that way, we do not have to update the highlightning rules when new pragmas or languages are introduced.